app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java

```diff
@@ -1,32 +1,211 @@
+    //And subscribe is used to check if User has already set a restaurant.
+    @SuppressWarnings({"ResultOfMethodCallIgnored", "ConstantConditions"})
+    @SuppressLint("CheckResult")
+    public void addRestaurant(Restaurant restaurant) {
+        if (currentUserMutableLiveData.getValue().getUserDailySchedule() != null) {
+            getRestaurantForKeyInteractor.getRestaurantForKey(currentUserMutableLiveData.getValue().getUserDailySchedule().getRestaurantKey())
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(restaurantsForKey -> {
+                        Restaurant originalChosenRestaurant = restaurantsForKey.get(0);
+                        if (originalChosenRestaurant.getName().equals(selectedRestaurantMutableLiveData.getValue().getName())) {
+                            Log.i("TAG", "addRestaurant: same restaurant");
+                        } else {
+                            checkIfRestaurantExistAndUpdateReservation(restaurant);
+                        }
+                    });
+        } else {
+            checkIfRestaurantExistAndUpdateReservation(restaurant);
+        }
+    }
+
+    //Suppress warning is safe cause subscribe is only used to set _currentUser
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @SuppressLint("CheckResult")
+    public void getCurrentUserForEmail(String authUserEmail) {
+        getCurrentUserForEmailInteractor.getCurrentUserForEmail(authUserEmail)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(users -> currentUserMutableLiveData.setValue(users.get(0)));
+    }
+
+    //Suppress warning is safe cause subscribe is used to check if restaurant exist in db
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @SuppressLint("CheckResult")
+    private void checkIfRestaurantExistAndUpdateReservation(Restaurant restaurant) {
+        getRestaurantForNameInteractor.getRestaurantForName(restaurant.getName())
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(restaurantsForName -> {
+                    if (restaurantsForName.isEmpty()) {
+                        addRestaurantWhenNotPersisted(restaurant);
+                    } else {
+                        Log.i("TAG", "onNext: restaurant exist" + restaurantsForName.get(0));
+                        updateUserForRestaurant(restaurant, restaurantsForName.get(0).getKey());
+                    }
+                });
+    }
+
+    private void addRestaurantWhenNotPersisted(Restaurant restaurant) {
+        restaurant.setRestaurantDailySchedule(new RestaurantDailySchedule(DATE, new ArrayList<>()));
+        addRestaurantInteractor.addRestaurant(restaurant)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
```

Comment:
Why subscribe isn't enough here as you've already managed to do with other observables ?

Reply:
Ok i didn't know that, if i understand, if my subscribeOn() and observeOn have been set in "parent" method that call other method, i don't have to set them again in "child" method?

Reply:
Done pass by flatMap, use andThen to combine observable on completable.

Find a more concise way to throw exeptions, just have to had a lambda throwable in subscribe.